evalml/tests/data_checks_tests/test_target_leakage_data_check.py

```diff
@@ -126,3 +128,282 @@ def test_target_leakage_data_check_input_formats():
+                                      message_code=DataCheckMessageCode.TARGET_LEAKAGE,
+                                      details={"column": "c"}).to_dict()],
+        "errors": []
+    }
+
+    # test X as ww.DataTable, y as ww.DataColumn
+    assert leakage_check.validate(ww.DataTable(X), ww.DataColumn(y)) == expected_messages
+
+    #  test y as list
+    assert leakage_check.validate(X, y.values) == expected_messages
+
+
+def test_target_leakage_regression():
+    leakage_check = TargetLeakageDataCheck(pct_corr_threshold=0.8)
+
+    # test empty pd.DataFrame, empty pd.Series
+    assert leakage_check.validate(pd.DataFrame(), pd.Series()) == {"warnings": [], "errors": []}
+
+    y = pd.Series([0.4, 0.1, 2.3, 4.3, 2.2, 1.8, 3.7, 3.6, 2.4, 0.9, 3.1, 2.8, 4.1, 1.6, 1.2])
+    X = pd.DataFrame()
+    X["a"] = y * 3
+    X["b"] = y - 1
+    X["c"] = y / 10
+    X["d"] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+    X["e"] = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o"]
+
+    expected_messages = {
+        "warnings": [DataCheckWarning(message="Column 'a' is 80.0% or more correlated with the target",
+                                      data_check_name=target_leakage_data_check_name,
+                                      message_code=DataCheckMessageCode.TARGET_LEAKAGE,
+                                      details={"column": "a"}).to_dict(),
+                     DataCheckWarning(message="Column 'b' is 80.0% or more correlated with the target",
+                                      data_check_name=target_leakage_data_check_name,
+                                      message_code=DataCheckMessageCode.TARGET_LEAKAGE,
+                                      details={"column": "b"}).to_dict(),
+                     DataCheckWarning(message="Column 'c' is 80.0% or more correlated with the target",
+                                      data_check_name=target_leakage_data_check_name,
+                                      message_code=DataCheckMessageCode.TARGET_LEAKAGE,
+                                      details={"column": "c"}).to_dict(),
+                     DataCheckWarning(message="Column 'e' is 80.0% or more correlated with the target",
+                                      data_check_name=target_leakage_data_check_name,
+                                      message_code=DataCheckMessageCode.TARGET_LEAKAGE,
+                                      details={"column": "e"}).to_dict()],
+        "errors": []
+    }
+
+    # test X as ww.DataTable, y as ww.DataColumn
+    assert leakage_check.validate(ww.DataTable(X), ww.DataColumn(y)) == expected_messages
+
+    #  test y as list
+    assert leakage_check.validate(X, y.values) == expected_messages
+
+
+def test_target_leakage_data_check_name():
+    y = pd.Series([1, 0, 1, 1])
+    X = pd.DataFrame()
+    X["target"] = y * 3
+    X["target0"] = y - 1
+    X["target00"] = y / 10
+    X["target000"] = ~y
+    X["e"] = [0, 0, 0, 0]
```

Comment:
I see that you decided to update the impl; in that case, do we still need these tests?

Reply:
Good catch, I'll remove these